lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -83,13 +82,16 @@ def _service(action: str, service_name: str, now: bool = None, quiet: bool = Non
         now: passes the --now flag to the shell invocation.
         quiet: passes the --quiet flag to the shell invocation.
     """
-    cmd = ["systemctl", action, service_name]
+    cmd = ["systemctl", cmd]
+    if service_name:
+        cmd.append(service_name)
+
     if now is not None:
         cmd.append("--now")
     if quiet is not None:
         cmd.append("--quiet")
-    if action != "is-active":
-        logger.debug("Attempting to {} '{}' with command {}.".format(action, service_name, cmd))
+    if cmd != "is-active":
+        logger.debug("Attempting to {} '{}' with command {}.".format(cmd, service_name, cmd))
```

Comment:
This check will fail now, because it is comparing against a lit, rather than a string. `if 'is-active' not in cmd` would work.